src/ansi-c/c_typecheck_initializer.cpp

```diff
@@ -663,10 +663,14 @@ void c_typecheck_baset::increment_designator(designatort &designator)
       assert(components.size()==entry.size);
 
       // we skip over any padding or code
-      while(entry.index<entry.size &&
+      // we also skip over anonymous members
+      while(entry.index < entry.size &&
             (components[entry.index].get_is_padding() ||
-             components[entry.index].type().id()==ID_code))
+             components[entry.index].get_anonymous() ||
```

Comment:
This causes Struct_Padding1 to fail, which contains an anonymous struct as part of the test. Should this test here be restricted to bitfields?